runtime/src/comm/gasnet/comm-gasnet-ex.c

```diff
@@ -1340,22 +1340,17 @@ void  chpl_comm_get_strd(void* dstaddr, size_t* dststrides, c_nodeid_t srcnode_i
   const size_t strlvls = (size_t)stridelevels;
   const gasnet_node_t srcnode = (gasnet_node_t)srcnode_id;
 
-  size_t dststr[strlvls];
-  size_t srcstr[strlvls];
-  size_t cnt[strlvls+1];
-
-  // Only count[0] and strides are measured in number of bytes.
-  cnt[0] = count[0] * elemSize;
+  ptrdiff_t dststr[strlvls];
+  ptrdiff_t srcstr[strlvls];
+  size_t cnt[strlvls];
```

Comment:
The VLA declarations above are only valid C when `strlvls > 0`. However this is also true of the old code, so I suspect this may be an undocumented invariant for `chpl_comm_{put,get}_strd()`? Might be worth an assertion to enforce that invariant.

FWIW GASNet Strided API _does_ correctly handle `strlvls == 0` as a contiguous transfer of `elemsz` bytes.

Reply:
Offline @benharsh confirmed that we do not call this code with `stridelevels == 0`. Agreed with the assert and subsequent code simplification.

Reply:
Hmm, I'm not sure that's true. The following code will trigger a strided GET with stridelvels=0

```chpl
config const stride = 1;
var A: [1..10] int;
var B: [1..10] int = 1..10;

A = B[1..10 by stride];
writeln(A);
```

So seems likely that the previous code was just buggy with the 0-length VLA. I'm going to pause this PR for now to make progress on some of the simpler changes.

Reply:
If you actually do need to support `stridelevels == 0`, the simplest way to solve this problem is probably to simply expand the VLA arrays by one element to ensure they are always non-empty:

```suggestion
  ptrdiff_t dststr[strlvls+1];
  ptrdiff_t srcstr[strlvls+1];
  size_t cnt[strlvls+1];
```

The extra element will be ignored by GASNet, and when `stridelevels == 0` all three arrays will be ignored entirely.

For completeness I'll note this is a C language compliance issue and not a GASNet one, although it really should be fixed to ensure portability. The analogous "extra VLA element" fix should probably also be applied to the GASNet-1 backend.